### ISSUE TITLE
[RFC]: add `fs/mkdir`

### ISSUE THREAD

**Issue by C0ldSmi1e** [stdlib-js/stdlib#2116]
### Description

This RFC proposes adding the package `@stdlib/fs/mkdir` for creating a directory asynchronously and synchronously.

Package: `@stdlib/fs/mkdir`
```javascript
var mkdir = require( '@stdlib/fs/mkdir' );
var mkdirSync = require( '@stdlib/fs/mkdir' ).sync;
```

### Related Issues

None.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

**Comment by C0ldSmi1e** [comment]
Hi @kgryte , can I work on this?

**Comment by kgryte** [comment]
@Daniel777y Yes, this would be useful. Note, however, that you'll want to accommodate older Node.js environments not supporting the `options` object (and hence `recursive`). You can consult https://github.com/isaacs/node-mkdirp, but note that the default API  for that package is now promise-based. You'll want to provide only a callback async API for `fs/mkdir`. Older versions of `node-mkdirp` were callback-based, but newer versions may have resolved some bugs. In short, you'll need to do some R&D to provide a polyfill for older Node.js versions.

**Comment by C0ldSmi1e** [comment]
> @Daniel777y Yes, this would be useful. Note, however, that you'll want to accommodate older Node.js environments not supporting the `options` object (and hence `recursive`). You can consult https://github.com/isaacs/node-mkdirp, but note that the default API for that package is now promise-based. You'll want to provide only a callback async API for `fs/mkdir`. Older versions of `node-mkdirp` were callback-based, but newer versions may have resolved some bugs. In short, you'll need to do some R&D to provide a polyfill for older Node.js versions.

Hi, @kgryte , just to clarity, I only need to align with the latest usage but to accommodate older Node.js environments, right? For example, `callback` was optional in older versions, but not passing it will still throw an error in `stdlib`. Correct me if I am wrong.

**Comment by kgryte** [comment]
Yes, I think that makes sense.
